jenkins/scripts/ironic_build_test.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+set -eu
+
+# shellcheck disable=SC1090
+source "${VARS_SCRIPT:-/tmp/vars.sh}"
+
+# This script is used to build Ironic and it is used by multiple different pipelines.
+# This script is designed to be executed on dynamically allocated VM-s provided
+# by the CI infrastrucutre.
+# Supported usecases:
+#   - Build ironic-image and install ironic from source
+#   - Build ironic-image and install ironic from rpm packages
+#   - Build ironic-image and install ironic from rpm packages then add a patch.
+
+IMAGE_REGISTRY="${IMAGE_REGISTRY:-localhost}"
+TAG="${TAG:-test}"
+CONTAINER_IMAGE_REPO="ironic-image"
+DESTINATION_REPO="${DETINATION_REPO:?}"
+DESTINATION_BRANCH="${DESTINATION_BRANCH:?}"
+SOURCE_REPO="${SOURCE_REPO:?}"
+SOURCE_BRANCH="${SOURCE_BRANCH:?}"
+PATCH_FILE=patch.txt
+OS_USERNAME="${OS_USERNAME:-metal3ci}"
+PATCHFILE_CONTENT="${PATCHFILE_CONTENT:-""}"
+IRONIC_INSTALL_MODE="${IRONIC_INSTALL_MODE:-INSTALL_FROM_RPM}"
+
+# This condition will be triggered for example when the related CI tooling (project-infra PR) is being
+# tested but not a PR in the ironic-image repository.
+
+# IMPORTANT: In case it is required to test the compatibility between project-infra and ironic-image PRs
+# then the test has to be triggered on the projec-infra PR then it should be manually stopped
+# and replayed in a way that the destination and source branches and repositories are manually modified
+# according to the inronic-image PR in question.
+# Example: user have in both project-infra and in ironic-image and the uswer would like to test both
+# together before any of them would be merged.
+if [[ "${DESTINATION_REPO}" != "https://github.com/metal3-io/ironic-image.git" ]];then
+    DESTINATION_REPO="https://github.com/metal3-io/ironic-image.git"
+    DESTINATION_BRANCH="main"
+    SOURCE_REPO="${DESTINATION_REPO}"
+    SOURCE_BRANCH="${DESTINATION_BRANCH}"
+fi
+
+case "$IRONIC_INSTALL_MODE" in
+
+  'INSTALL_FROM_SOURCE')
```

Comment:
maybe don't repeat `prepare_work_env` take it out of case?

Reply:
It is done this way to avoid work environment preparation in case unsupported IRONIC_INSTALL_MODE mode was supplied to the script.

Reply:
For that you could add a check to see if the user has provided unsupported method and fail early

Reply:
This is the earliest part in this script where any actual "logic" is implemented. If you mean that I could do it outside this script then you are right but I have implemented the script to be usable in a standalone mode in case somebody has to quickly test something on local machine.